### ISSUE TITLE
Provision: slurm for HPC environments

### ISSUE THREAD

**Issue by LecrisUT** [teemtee/tmt#2780]
In HPC a common standard is to use `slurm` to "provision" test workers, especially for highly parallelized or specific hardware requirements. I got a suggestion for having this kind of `provision` step, and for the most part I think it would be doable.

There is one issue though, there is no `ssh` like interface to perform shell commands. `salloc` could be used to prepare the host, and maybe `sattach` could be used to add new step there.

**Comment by lukaszachy** [comment]
We don't need `ssh` per se, do you see problems in having own guest class for slum? https://tmt.readthedocs.io/en/stable/code/autodocs/tmt.steps.provision.html#tmt.steps.provision.Guest

**Comment by LecrisUT** [comment]
Indeed, that could work. Need to setup a proper multi-guest tmt environment to test that out because there are various edge cases in that `srun` could potentially run in heterogeneous environments, most commonly, there could be `/scratch`, `/tmp` folders that are not shared.

**Comment by LecrisUT** [comment]
Nice to have, but would not focus on it. If more relevant in the future will re-open
